projectile.el

```diff
@@ -4069,10 +4069,12 @@ project of that type"
 
 (defun projectile-read-command (prompt command)
   "Adapted from `compilation-read-command'."
-  (read-shell-command prompt command
-                      (if (equal (car compile-history) command)
-                          '(compile-history . 1)
-                        'compile-history)))
+  (let ((compile-history
+         (ring-elements (projectile--get-command-history (projectile-ensure-project (projectile-project-root))))))
```

Comment:
I don't quite get how this is supposed to work, so can you explain it in more details?

Reply:
Yup, I see the package has `projectile--get-command-history` function. It provides command history per project.

I would like to separate the command history per project. So I tried to set `compile-history` value from `projectile--get-command-history` (I think it as  reference/a pointer). And I see it works. I ever did `xref--marker-ring` with the same idea. 

If you ask me in detail how it works, I don't think I know the deep

Reply:
I was mostly curious why do you need `ring-elements` here, but I assume you copied it from somewhere else in the code.

Reply:
Yes, you're right. I copied from here https://github.com/bbatsov/projectile//blob/7b6ac286120421216faae43f9e234d801a30ee3d/projectile.el#L4231